exercises/csharp/TriviaGame/TriviaGame/Game.cs

```diff
@@ -2,190 +2,194 @@
+            }
+            return true;
+        }
+        public void AdvanceOnGameBoard(int roll)
+        {
+            places[currentPlayer] = places[currentPlayer] + roll;
         }
 
-        public void Roll(int roll)
+        public void RollDice(int roll)
         {
-            Console.WriteLine(players[currentPlayer] + " is the current player");
-            Console.WriteLine("They have rolled a " + roll);
+            Console.WriteLine($"{players[currentPlayer]} is the current player");
+            Console.WriteLine($"They have rolled a {roll}");
 
             if (inPenaltyBox[currentPlayer])
             {
                 if (roll % 2 != 0)
                 {
                     isGettingOutOfPenaltyBox = true;
 
-                    Console.WriteLine(players[currentPlayer] + " is getting out of the penalty box");
-                    places[currentPlayer] = places[currentPlayer] + roll;
-                    if (places[currentPlayer] > 11) places[currentPlayer] = places[currentPlayer] - 12;
+                    Console.WriteLine($"{players[currentPlayer]} is getting out of the penalty box");
 
-                    Console.WriteLine(players[currentPlayer]
-                            + "'s new location is "
-                            + places[currentPlayer]);
-                    Console.WriteLine("The category is " + CurrentCategory());
+                    AdvanceOnGameBoard(roll);
+                    IsBackToStartOfBoard();
+
+                    Console.WriteLine($"{players[currentPlayer]}'s new location is {places[currentPlayer]}");
+                    Console.WriteLine($"The category is {Category()}");
                     AskQuestion();
                 }
                 else
                 {
-                    Console.WriteLine(players[currentPlayer] + " is not getting out of the penalty box");
+                    Console.WriteLine($"{players[currentPlayer]} is not getting out of the penalty box");
                     isGettingOutOfPenaltyBox = false;
                 }
-
             }
             else
             {
+                AdvanceOnGameBoard(roll);
+                IsBackToStartOfBoard();
 
-                places[currentPlayer] = places[currentPlayer] + roll;
-                if (places[currentPlayer] > 11) places[currentPlayer] = places[currentPlayer] - 12;
-
-                Console.WriteLine(players[currentPlayer]
-                        + "'s new location is "
-                        + places[currentPlayer]);
-                Console.WriteLine("The category is " + CurrentCategory());
+                Console.WriteLine($"{players[currentPlayer]}'s new location is {places[currentPlayer]}");
+                Console.WriteLine($"The category is {Category()}");
                 AskQuestion();
             }
 
         }
 
         private void AskQuestion()
         {
-            if (CurrentCategory() == "Pop")
+            if (Category() == "Pop")
```

Comment:
These strings could be extracted to constant fields or an enum so that if someone misspells a word, or uses lower case by accident "pop", then it won't break